aderyn_pilot/templates/detector_rs.hbs

```diff
@@ -0,0 +1,72 @@
+use std::{collections::BTreeMap, error::Error};
+
+use aderyn_driver::context::loader::ContextLoader;
+use aderyn_driver::detection_modules::capture;
+use aderyn_driver::detector::{Detector, IssueSeverity};
+
+#[derive(Default)]
+pub struct {{detector_name_title_case}} {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize), String>,
+}
+
+impl Detector for {{detector_name_title_case}} {
+    fn detect(&mut self, loader: &ContextLoader) -> Result<bool, Box<dyn Error>> {
+        
+        // Use the `loader` to find nodes, then capture them as shown below
+        // capture!(self, loader, event_definition);
+        
+        Ok(!self.found_instances.is_empty())
+    }
+
+    fn title(&self) -> String {
+        String::from("Title for {{ detector_name_title_case }}")
+    }
+
+    fn description(&self) -> String {
+        String::from("Description for {{ detector_name_title_case }}")
+    }
+
+    fn severity(&self) -> IssueSeverity {
+        // Choose the appropriate severity
+        IssueSeverity::NC
+    }
+
+    fn instances(&self) -> BTreeMap<(String, usize), String> {
+        self.found_instances.clone()
+    }
+}
+
+#[cfg(test)]
+mod {{detector_name_camel_case}}_tests {
+
+    use crate::config_tests::tests_configuration;
+
+    use super::{{detector_name_title_case}};
+
+    use aderyn_driver::context::loader::ContextLoader;
+    use aderyn_driver::detector::detector_test_helpers::load_contract;
+    use aderyn_driver::detector::Detector;
+
+    fn test_{{detector_name_camel_case}}_for(
```

Comment:
this test function will run against all the test files that have this detector in their `with`  clause